src/components/Dropdown/Dropdown.tsx

```diff
@@ -0,0 +1,107 @@
+import React, { ReactElement } from "react";
+import styled, { css } from "styled-components";
+import { Menu } from "@headlessui/react";
+import { IconDirection } from "../Icon/Direction";
+import { Text } from "../../typography";
+import { Button } from "../Button";
+import Icon from "../Icon";
+
+export interface DropdownOption {
+  value: string;
+  label: string;
+}
+
+export interface DropdownProps {
+  options: DropdownOption[];
+  defaultOption: DropdownOption;
+  selected?: DropdownOption;
+  setSelected: (option: DropdownOption) => void;
+}
+
+export const Dropdown = ({
+  options,
+  defaultOption,
+  selected,
+  setSelected,
+}: DropdownProps): ReactElement => {
+  const isDropdownOptionSelected = ({ value }: DropdownOption): boolean => {
+    return value === selected?.value;
+  };
+
+  return (
+    <Menu>
+      {({ open }) => (
+        <>
+          <StyledMenuButton>
+            <StyledLabel>{selected?.label || defaultOption.label}</StyledLabel>
+            <ChevronComponent open={open} />
+          </StyledMenuButton>
+          <StyledMenuItems>
+            {options.map((option) => (
+              <Menu.Item key={option.value} onClick={() => setSelected(option)}>
+                {({ active }) => (
+                  <StyledMenuItem
+                    active={active || isDropdownOptionSelected(option)}
+                  >
+                    {option.label}
+                  </StyledMenuItem>
+                )}
+              </Menu.Item>
+            ))}
+          </StyledMenuItems>
+        </>
+      )}
+    </Menu>
+  );
+};
+
+const ChevronComponent = ({ open }: { open: boolean }): ReactElement => {
+  const direction = open ? IconDirection.Up : IconDirection.Down;
+
+  return (
+    <Button.Icon icon={<Icon.Chevron direction={direction} size={16} />} />
+  );
+};
+
+const StyledMenuButton = styled(Menu.Button)`
+  align-items: center;
+  background: ${({ theme }) => theme.colors.backgroundVariant};
+  border-radius: 0.5rem;
+  border: 0;
+  color: ${({ theme }) => theme.colors.primary};
+  display: flex;
+  height: 2rem;
+  justify-content: space-between;
+  cursor: pointer;
+  padding-left: 0.75rem;
+`;
+
+const StyledLabel = styled(Text)`
+  color: ${({ theme }) => theme.colors.primary};
+`;
+
+const StyledMenuItems = styled(Menu.Items)`
+  background: ${({ theme }) => theme.colors.backgroundVariant};
+  border-radius: 0.5rem;
```

Comment:
I would not put border-radius here
```suggestion
  border-radius: 0rem;
```

Reply:
Are we sure? The designs looked different

<img width="331" alt="Screen Shot 2022-03-14 at 9 28 12 am" src="https://user-images.githubusercontent.com/78794805/158081866-2ba5ec41-3b30-44cf-b597-d07cdc4b190a.png">

Design:

<img width="213" alt="Screen Shot 2022-03-14 at 9 28 44 am" src="https://user-images.githubusercontent.com/78794805/158081882-3c0585ef-21be-4e02-a41c-fdd7766770f5.png">

Reply:
As discussed, fixed with https://github.com/Major-Eth/yearn-web-lib/pull/28/commits/cfc35348e2b5cb52a9ee5b7e7204aac800a86a6b

So now only the corners have radius and when hover the color of the text remains blue.